tests/core/pyspec/eth2spec/test/bellatrix/confirmation_rule/test_confirmation_rule.py

```diff
@@ -0,0 +1,585 @@
+        self.confirmation_byzantine_threshold = confirmation_byzantine_threshold
+        self.confirmation_slashing_threshold = confirmation_slashing_threshold
+
+    def check_is_confirmed(self, store, block_root, test_steps, expected=None):
+        confirmed = self.spec.is_confirmed(
+            store, self.confirmation_byzantine_threshold, self.confirmation_slashing_threshold, block_root
+        )
+        if expected is not None:
+            assert confirmed == expected
+        test_steps.append({"check_is_confirmed": {"result": confirmed, "block_root": str(block_root)}})
+
+    def is_lmd_confirmed(
+        self,
+        store,
+        block_root,
+    ):
+        return self.spec.is_lmd_confirmed(store, self.confirmation_byzantine_threshold, block_root)
+
+    def is_ffg_confirmed(
+        self,
+        store,
+        block_root,
+    ):
+        return self.spec.is_ffg_confirmed(
+            store, self.confirmation_byzantine_threshold, self.confirmation_slashing_threshold, block_root
+        )
+
+    def check_get_confirmation_score(self, store, block_root, test_steps, expected=None):
+        confirmation_score = int(
+            self.spec.get_confirmation_score(store, self.confirmation_slashing_threshold, block_root)
+        )
+        if expected is not None:
+            assert confirmation_score == expected
+        test_steps.append(
+            {"check_get_confirmation_score": {"result": confirmation_score, "block_root": str(block_root)}}
+        )
+
+    def get_confirmation_score(self, store, block_root):
+        return self.spec.get_confirmation_score(store, self.confirmation_slashing_threshold, block_root)
+
+    def get_lmd_confirmation_score(self, store, block_root):
+        return self.spec.get_lmd_confirmation_score(store, self.confirmation_slashing_threshold, block_root)
+
+    def get_ffg_confirmation_score(self, store, block_root):
+        return self.spec.get_lmd_confirmation_score(store, self.confirmation_slashing_threshold, block_root)
+
+
+def confirmation_rule_setup(confirmation_byzantine_threshold: int, confirmation_slashing_threshold: int):
+    def decorator(func):
+        def wrapped(*args, **kwargs):
+            yield "setup", {
+                "confirmation_byzantine_threshold": confirmation_byzantine_threshold,
+                "confirmation_slashing_threshold": confirmation_slashing_threshold,
+            }
+            yield from func(
+                *args,
+                conf_rule_wrapper=ConfirmationRuleWrapper(
+                    spec=kwargs["spec"],
+                    confirmation_slashing_threshold=confirmation_slashing_threshold,
+                    confirmation_byzantine_threshold=confirmation_byzantine_threshold,
+                ),
+                **kwargs
+            )
+
+        return wrapped
+
+    return decorator
+
+
+@with_all_phases
+@spec_state_test
+@confirmation_rule_setup(confirmation_byzantine_threshold=0, confirmation_slashing_threshold=0)
```

Comment:
Once you moved these two settings to `Configurations`, you can use the `@with_config_overrides()`  decorator to set config per test case. It will yield the config yaml file automatically.

Reply:
Done